### PR TITLE
[MISC] Update search_cfg::max_error_deletion to recent config design changes.

### PR DESCRIPTION
**Pull request by rrahn**
part of seqan/product_backlog#217

### CODE REVIEW

**Review by eseiler — APPROVED** [review]

File: include/seqan3/search/configuration/max_error.hpp [hunk 1]
```diff
@@ -89,9 +89,34 @@ class max_error_insertion : public pipeable_config_element<max_error_insertion,
  * ### Example
  * \include test/snippet/search/configuration_error.cpp
  */
-class max_error_deletion : public pipeable_config_element<max_error_deletion, std::variant<error_count, error_rate>>
+struct max_error_deletion : public pipeable_config_element<max_error_deletion>
 {
-public:
+    //!\brief  The number of deletion errors allowed for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:94]:
```suggestion
    //!\brief  The number of deletion errors or the deletion error rate, allowed for the search.
```

File: include/seqan3/search/configuration/max_error.hpp [hunk 2]
```diff
@@ -89,9 +89,34 @@ class max_error_insertion : public pipeable_config_element<max_error_insertion,
...
+     */
+    max_error_deletion() = default; //!< Defaulted.
+    max_error_deletion(max_error_deletion const &) = default; //!< Defaulted.
+    max_error_deletion(max_error_deletion &&) = default; //!< Defaulted.
+    max_error_deletion & operator=(max_error_deletion const &) = default; //!< Defaulted.
+    max_error_deletion & operator=(max_error_deletion &&) = default; //!< Defaulted.
+    ~max_error_deletion() = default; //!< Defaulted.
+
+    /*!\brief Initialises the deletion error with the given seqan3::search_cfg::error_count.
+     * \param[in] error The error count used for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:108]:
```suggestion
     * \param[in] error The number of deletion errors allowed in the search.
```

File: include/seqan3/search/configuration/max_error.hpp [hunk 3]
```diff
@@ -89,9 +89,34 @@ class max_error_insertion : public pipeable_config_element<max_error_insertion,
...
+     */
+    explicit max_error_deletion(error_count error) : error{std::move(error)}
+    {}
+
+    /*!\brief Initialises the deletion error with the given seqan3::search_cfg::error_rate.
+     * \param[in] error The error rate used for the search.
```
Comment (smehringer) [inline include/seqan3/search/configuration/max_error.hpp:114]:
```suggestion
     * \param[in] error The deletion error rate allowed in the search.
```

**Review by smehringer — APPROVED** [review]
only dox. after that LGTM

File: include/seqan3/search/configuration/max_error.hpp (hunk 1 above)
Comment (rrahn) [reply include/seqan3/search/configuration/max_error.hpp:94]: I'd propose: `The error count or error rate.` That it is meant for the search is already clear by the class's documentation.

**Pull request merged**